### PR TITLE
podman create: building local pause image: do not read ignore files

### PR DESCRIPTION
**Pull request by vrothberg**
Make sure to ignore local {container,docker}ignore files when building a
local pause image.  Otherwise, we may mistakenly not be able to copy
catatonit into the build container.

Fixes: #13529
Signed-off-by: Valentin Rothberg <vrothberg@redhat.com>

### CODE REVIEW

File: test/system/070-build.bats
Comment (vrothberg) [reply test/system/070-build.bats:625]: Being sloppy justifies remote slaps ;-) Thanks for checking.  I was too sure it's doing exactly what I thought.

**Comment by vrothberg** [comment]
@edsantiago updated. I tried making up by consolidating places using the local pause image.

**Comment by vrothberg** [comment]
@containers/podman-maintainers PTAL

/hold
... as I want a final head nod from Ed.

**Review by Luap99 — COMMENTED** [review]
LGTM

File: test/system/helpers.bash
Comment (edsantiago) [inline test/system/helpers.bash:389]: This looks like a NOP, but there might be circumstances in which the version includes a slash. If there are, could you add a comment explaining what that's all about? Or if it truly is a NOP, could you remove this line?

(also: if you need to keep this line, could you rename the variable to `tag`, which seems more accurate?)

File: test/system/070-build.bats
Comment (edsantiago) [inline test/system/070-build.bats:628]: Thanks! No real need for a variable, though: could just be `run_podman rmi $(pause_image)` like you did below.

File: test/system/070-build.bats
Comment (edsantiago) [inline test/system/070-build.bats:623]: Because this looks like a NOP (pod create + rm with no actual testing of output), would you mind expanding on the comment? Something like "Prior to this fix, pod-create would fail with 'error building at STEP COPY .../catatonit' because of the ignore pattern". That would save a future maintainer having to look at the issue.

**Review by edsantiago — COMMENTED** [review]
LGTM, and the test now fails (as it should) in unpatched podman, but a few suggestions inline.

Thank you!

File: test/system/200-pod.bats
Comment (edsantiago) [inline test/system/200-pod.bats:9]: One more concern: this `rmi` will fail if `pause_image` does not exist, e.g., if some test fails in a step before creating a pod. Unlikely, but if it does happen, `teardown` will fail, and that can make a mess. With a 'grep images' approach, a failure is less likely.

I'm not sure what to recommend here. One approach would be to add `run_podman rmi $(pause_image)` to each subtest. I like that, because it moves cleanup into each test, but it's probably more than you signed up for in this PR. 

I tentatively suggest just forgetting about this part of the PR, and letting me file a new PR later that will handle cleanup. I need to do that anyway, because we have some ugly-read-yuk right now because one of these tests uses a `k8s.something/pause` which this teardown does not clean up.

Comment (vrothberg) [reply test/system/200-pod.bats:9]: Great catch!  We already have an `--ignore` flag for `container rm` and this screams like we should have one for `image rm` as well.

File: test/system/helpers.bash
Comment (vrothberg) [reply test/system/helpers.bash:388]: Without, it doesn't work as `run_podman` does some magic.  I copied it from another function above.

File: test/system/helpers.bash
Comment (vrothberg) [reply test/system/helpers.bash:389]: Great catch!

File: test/system/helpers.bash
Comment (edsantiago) [reply test/system/helpers.bash:388]: That seems really astonishing. Since it goes against almost everything I know about BATS, could you add a comment explaining why it's needed? (With a little more detail than "doesn't work"?) That is just really unexpected.

File: test/system/200-pod.bats
Comment (vrothberg) [reply test/system/200-pod.bats:9]: `run_podman ? rmi` could work

File: test/system/helpers.bash
Comment (vrothberg) [reply test/system/helpers.bash:388]: `git blame` brought me to one of your commit 9fd7ab50f82c7:
>  This function is intended to be used as '$(podman_runtime)', i.e.
>  our caller wants our output. run_podman() messes with output because
> it emits the command invocation to stdout, hence the redirection.

**Review by edsantiago — APPROVED** [review]
/lgtm
/hold

**Pull request merged**
